v3_async_wip/v3_async_wip/iothub_mqtt_client.py

```diff
@@ -153,19 +153,19 @@ async def shutdown(self) -> None:
         # TODO: this breaks when called twice. Build some protections.
         # TODO: is there an issue with cancellation here?
         await self.disconnect()
-        # Cancel the SAS token update alarm. Note that this is not a task, it's a threaded Alarm.
-        # No need to wait for the result.
-        if self._sastoken_update_alarm:
-            logger.debug("Cancelling SAS Token update alarm")
-            self._sastoken_update_alarm.cancel()
-        # Cancel and wait for the completion of the twin response task
-        logger.debug("Cancelling twin response listener")
-        self._twin_response_listener.cancel()
+        cancelled_tasks = []
```

Comment:
>     cancelled_tasks = [] [](http://example.com/codeflow?start=3&length=25)

Recommend wrapping self.disconnect in a try/catch block.

Reply:
Disconnect can't fail. I could add one for safety, I suppose, but it would be purely defensive.

Reply:
but it can fail, it calls logger.debug and it calls into Paho, and either of those could raise an exception for any number of reasons that are beyond our control.  Yes, unlikely, but this function gets called when things are getting torn down or things are broken, so it's hard to say what can fail and what can't. 

Normally I wouldn't worry about being defensive like this, but `shutdown` is special.  It's an "I need to get out of here" function, and we don't want it to fail. It's like the "power off" button on your phone. In normal situations, nothing should go wrong with powering off your phone, but you don't want the "power off" button to fail and leave your phone partially powered on with no recourse.

Reply:
Fair enough on the point about logging, I didn't consider the implications of that. So would you like to see it suppress the error completely, or cache the error, continue trying to shut other things down, and then raise the error at the end?

Reply:
Good question. I was thinking we should suppress, but re-raising at the end is an interesting idea. Not sure which is better though.